onnxruntime/core/providers/cuda/reduction/reduction_ops.cc

```diff
@@ -284,16 +285,34 @@ Status PrepareForReduce(const Tensor* X,
   std::vector<bool> reduced(rank, false);
   prepare_reduce_metadata.output_dims.reserve(input_dims.size());
   if (axes.size() > 0) {
+    int64_t reduced_axis;
+    std::vector<uint64_t> axes_(axes.size());
     prepare_reduce_metadata.output_dims = input_dims;
-    for (auto reduced_axis : axes) {
+    for (size_t i = 0; i < axes.size(); i++) {
+      reduced_axis = axes[i];
       const int64_t axis = HandleNegativeAxis(reduced_axis, rank);
-      ORT_ENFORCE(axis < rank, "Reduced axis is greater than rank: ", axis);
       ORT_ENFORCE(input_dims[axis] != 0,
                   "Can't reduce on dim with value of 0 if 'keepdims' is false. "
                   "Invalid output shape would be produced. input_shape:",
                   input_shape);
       prepare_reduce_metadata.output_dims[axis] = 1;
       reduced[axis] = true;
+      axes_[i] = axis;
+    }
+
+    bool contiguous_axes = true;
+    std::sort(axes_.begin(), axes_.end());
```

Comment:
probably safer to make this a set<int64_t> and just compare if current == prev+1, you can get rid of the sort.

Reply:
the axes must start with 0 along with being current=prev+1. Since we don't support rank more than 8, this is a very cheap sort.